+nosnoc/+solver/MpccSolver.m

```diff
@@ -426,5 +426,443 @@ function print_infeasibility(obj, results)
                 keyboard
             end
         end
+
+        function [polished_w, res_out, stat_type, n_biactive] = calculate_stationarity(obj, results, exitfast, lifted)
+            import casadi.*
+            stat_type = "?";
+            nlp = obj.nlp;
+            mpcc = obj.mpcc;
+
+            % This method generates a TNLP (tightened NLP) using the results of a homotopy solver.
+            % In order to do so we first identify the active set of the complementarity constraints which involves identifying which
+            % branch of the complementarity each element of G(w) and H(w) are, or whether the solution is biactive at that point.
+            % For a more in depth explanation see the PhD theses of Armin Nurkanovic or Alexandra Schwartz
+            
+            w_orig = results.x(nlp.ind_map);
+            lam_x = full(results.lam_x(nlp.ind_map));
+            lam_g = full(results.lam_g);
+            lam_g(nlp.ind_g_comp) = []; % TODO: also elastic scholtes constraint
```

Comment:
what is this? I don' understand what is meant by elastic scholtes constraints and how it is relevant?

Reply:
I believe you are looking at an old commit here. This no longer exists.